### DISCUSSION TITLE
Can't install Perspective: Could not build wheels for perspective-python

### DISCUSSION THREAD

**Discussion by fawazghali** [perspective-dev/perspective#1884] [Q&A] [answered]
Trying to install `perspective-python`. We’re working on a presentation for an upcoming FINOS talk in London in July 2022
 
`pip 22.1.2 from /opt/homebrew/lib/python3.9/site-packages/pip (python 3.9)`

```
full command: /opt/homebrew/opt/python@3.9/bin/python3.9 /opt/homebrew/lib/python3.9/site-packages/pip/_vendor/pep517/in_process/_in_process.py build_wheel /var/folders/dr/vvyk791n18v0cny3_9bwvsmw0000gp/T/tmpz49re9wk
  cwd: /private/var/folders/dr/vvyk791n18v0cny3_9bwvsmw0000gp/T/pip-install-kwdhd71h/perspective-python_4fa2f8c70e514f35a69489fa0e09bd46
  Building wheel for perspective-python (pyproject.toml) ... error
  ERROR: Failed building wheel for perspective-python
Failed to build perspective-python
ERROR: Could not build wheels for perspective-python, which is required to install pyproject.toml-based projects

```

**Complete log file**
[perspective-python.txt](https://github.com/finos/perspective/files/8851733/perspective-python.txt)

**Comment by timkpaine** [comment]
We don't publish non x86 Mac wheels yet as the CI resources are not available. We do build Mac arm wheels via conda forge.

For your error, try removing your homebrew-installed flatbuffers and retry the installation.

**Comment by timkpaine** [comment, marked as answer]
starting with python 3.11, we now publish universal binaries for mac.

**Comment by eo1989** [comment]
Not to resurface a dead thread, but just tried installing my requirements.txt in a 3.12 virtualenv and perspective-python fails to build. What is odd is if pip install --dry-run (w/ || w/o --pre) perspective-{python,parquet} the command returns "Would install perspective-python-2.8.0 perspective-parquet-0.1.1". To be clear, it is in wsl2 ubuntu 22.04 && the flatbuffers is not installed via homebrew.
[perspective-log.txt](https://github.com/finos/perspective/files/14323830/perspective-log.txt)

**Reply by timkpaine** [reply]
your locally-installed `re2` is conflicting with our vendored `re2` as indicated in the error messages
